### PR TITLE
feat(analytics): Adding default event parameters

### PR DESCRIPTION
**Pull request by MFrat**
### Description
Adding feature for setting default event parameters. As described [here](https://firebase.google.com/docs/reference/kotlin/com/google/firebase/analytics/FirebaseAnalytics#setdefaulteventparameters) in Firebase docs.

### Release Summary

Added `setDefaultEventParameters`. It adds default event parameters that will be send in every event.

### Checklist

- I read the [Contributor Guide](../CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
  - [x] Yes
- My change supports the following platforms;
  - [x] `Android`
  - [x] `iOS`
- My change includes tests;
  - [x] `e2e` tests added or updated in `packages/\*\*/e2e`
  - [x] `jest` tests added or updated in `packages/\*\*/__tests__`
- [x] I have updated TypeScript types that are affected by my change.
- This is a breaking change;
  - [ ] Yes
  - [x] No

### Test Plan

---

Think `react-native-firebase` is great? Please consider supporting the project with any of the below:

- 👉 Star this repo on GitHub ⭐️
- 👉 Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter

### CODE REVIEW

**Review by mikehardy — APPROVED** [review]
Wow, this looks fantastic first shot. Thank you!
Github requires that a maintainer approves the CI runs on first contributions (to avoid PRs that just implement a cryptominer I think) - so I've done that and will wait for CI to go green

Looks like other than that you just need to follow the "Details" link on the failing CLA / license check and sign the license then we can merge this

**Comment by mikehardy** [comment]
Oh - now that CI has run it looks like some trivial formatting issues, and the jest error message isn't quite as expected as well as the CLA - waiting for those

**Comment by MFrat** [comment]
Thanks for the tips. Done!

**Comment by mikehardy** [comment]
Looks like there is still an issue with the CLA signing - I have had this happen before where I needed to add the email address as it mentions:

> Max Fratane seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, [please add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).
You have signed the CLA already but the status is still pending? Let us recheck it.


It appears to be looking for yoru max2fratane @ popular mail provider email address as that's the address in the commits on your fork

**Comment by MFrat** [comment]
E-mail added to my github account.

File: packages/analytics/ios/RNFBAnalytics/RNFBAnalyticsModule.m
Comment (mikehardy) [inline packages/analytics/ios/RNFBAnalytics/RNFBAnalyticsModule.m:153]: I think the API call needs to look like this - and I *think* I can commit the suggestion so CI will chew on it and validate my assertion

```suggestion
      [FIRAnalytics setDefaultEventParameters:[self cleanJavascriptParams:params]];
```

File: packages/analytics/ios/RNFBAnalytics/RNFBAnalyticsModule.m
Comment (mikehardy) [inline packages/analytics/ios/RNFBAnalytics/RNFBAnalyticsModule.m:146]: And I believe the API definition should look like this - I am able to commit these, then CI chews through...
```suggestion
```

**Comment by mikehardy** [comment]
Hey @MFrat - before I merge this, since it seemed like the iOS side had compile errors and couldn't have been checked - can you install the auto-generated [patch-package format patches](https://github.com/ds300/patch-package) from CI and make sure they work in your project, with expected results on android and iOS? There is an artifacts link here that has them https://github.com/invertase/react-native-firebase/actions/runs/803147844

**Comment by MFrat** [comment]
@mikehardy, I tested both in Android and iOS, with devices and simulator/emulator. Everything is working just fine.

Screen Shot 2021-05-02 at 16 01 38
Screen Shot 2021-05-02 at 15 43 02

**Android**
Screen Shot 2021-05-02 at 15 59 34

**iOS**
Screen Shot 2021-05-02 at 16 29 31
Screen Shot 2021-05-02 at 16 28 21

**Comment by mikehardy** [comment]
That's glorious :-), patch-package is such a beautiful thing. Thanks again for the PR here I'll merge it but hold for a few days prior to release in order to catch any other changes - you're set via patch-package until then though (it gives you a nice warning when you npm/yarn install after the package it has a patch for changes version so you'll know to clear the patch from source control later)

**Pull request merged**
